lumigator/python/mzai/backend/backend/services/datasets.py

```diff
@@ -83,6 +84,15 @@ def validate_experiment_dataset(filename: str):
             )
 
 
+def dataset_has_gt(filename: str) -> bool:
+    with Path(filename).open() as f:
+        reader = csv.DictReader(f)
+        fields = set(reader.fieldnames or [])
+        has_gt = GT_FIELD in fields
```

Comment:
I'd argue that here the semantics is a bit different, i.e. we are validating the existence of a field rather than rows being consistent with a schema (see examples below).

However, I think the idea is neat and I'd like to include it into another PR. I'd extend this to the whole dataset (instead of just the first row) and use to make sure all of our data is consistent with the schema we provide. WDYT?

---

This is the code I tested with:

```
class CSVFields(BaseModel):
    examples: str
    ground_truth: str | None = None


def validate_ground_truth(filename) -> bool:
    has_ground_truth = False

    with Path(filename).open() as f:
        reader = csv.DictReader(f)
        first_row = next(reader, None)
        logger.info(f"First row: {first_row}")
        if first_row:
            try:
                validated_row = CSVFields.model_validate(first_row)
                logger.info(f"Validated row: {validated_row}")
                has_ground_truth = validated_row.ground_truth is not None
            except ValidationError as e:
                print(f"Validation error: {e}")

    logger.info(f"has_ground_truth: {has_ground_truth}")
    return has_ground_truth
```

And this what I get with different inputs:

INPUT (no gt):
```
examples
"This is an example"
"This is another example"
```
OUTPUT:
```
First row: {'examples': 'This is an example'}
Validated row: examples='This is an example' ground_truth=None
has_ground_truth: False
```

---


INPUT (gt field, one row without gt):
```
examples,ground_truth
"This is an example", "This is a summary"
"This is another example"
```
OUTPUT:
```
First row: {'examples': 'This is an example', 'ground_truth': ' "This is a summary"'}
Validated row: examples='This is an example' ground_truth=' "This is a summary"'
has_ground_truth: True
```

---

INPUT (same file as before, flipped rows)
```
examples,ground_truth
"This is another example"
"This is an example", "This is a summary"
```

OUTPUT:
```
First row: {'examples': 'This is another example', 'ground_truth': None}
Validated row: examples='This is another example' ground_truth=None
has_ground_truth: False
```

---

INPUT (same file as before, one extra comma)
```
examples,ground_truth
"This is another example",
"This is an example", "This is a summary"
```

OUTPUT:
```
First row: {'examples': 'This is another example', 'ground_truth': ''}
Validated row: examples='This is another example' ground_truth=''
has_ground_truth: True```

---

Reply:
Sounds good 👍